doc/rst/technotes/initializers.rst

```diff
@@ -12,123 +12,289 @@ than the original methods known as constructors.
+preceded with the ``new`` keyword.
 
-  - user-defined code that runs in phase 2
+If the program declares a type initializer method, it is a user-defined
+initializer.  If the program declares no initializers or constructors for a
+class or record, a compiler generated constructor for that type is created
+automatically (see below for the status on `Compiler Generated Initializers`_).
 
-  - user-defined code that runs in phase 1
+User-Defined Initializers
+-------------------------
 
-  - compiler generated code that runs in phase 1
+A user-defined initializer is an initializer method explicitly declared in the
+program.  An initializer declaration has the same syntax as a method
+declaration, except that the name of the function is "init", and there is no
+return type specifier.
 
-  - the generation of effective user-facing diagnostic messages
+When an initializer is called, the usual function resolution mechanism is
+applied to determine which user-defined initializer to invoke.
 
-in approximate order of robustness and maturity.
+In contrast to user-defined constructors for generic classes and records,
+user-defined initializers do not require an argument per generic field.
 
-The user-defined code that executes in phase 2 of an initializer is
-comparable to the user-defined code in any method and is likely to
-behave in the expected way.
 
-The user-defined code that executes in phase 1 of an initializer has
-many of the same properties and is also expected to behave well.
-However there are more restrictions on user-defined code for phase 1
-and the compiler may fail to apply these restrictions correctly.  The
-most likely error is that the compiler will accept phase 1 code that
-it should reject but it is possible that there will be situations in
-which valid code is rejected.
 
-The design for initializer methods attempts to define predictable
-default behaviors within phase 1 and then rely on the compiler to
-insert these defaults when appropriate.  The intent is to reduce the
-amount of code that the type designer is responsible for.  This
-includes the generation of code using field-level default
-declarations.  Defining and implementing these default behaviors
-relies on relatively less mature code within the compiler.
+The Initializer Body
+--------------------
 
-We believe that initializer methods are ready for use in application code
-but we also expect that the implementation will mature significantly
-during the 1.15 release.  We encourage their use but recognize that they
-may prove to be less robust than we would like at the start of the release.
-If you do choose to experiment with this feature then please do not
-hesitate to seek help, report bugs, or offer suggestions.
+The code written in an initializer can be divided into two categories,
+referred to as "Phase 1" and "Phase 2" for the remainder of this document.
+When the phase division indicator is not present, the body of an initializer
```

Comment:
I wonder if there should be a summary of what phase1/phase2 are, and perhaps even a summary of the motivation, before talking about the "division indicator" etc?